plugins/nomad/src/api.ts

```diff
@@ -0,0 +1,160 @@
+  ) => Promise<ListAllocationsResponse>;
+
+  /**
+   * listJobVersions is for listing all deployments matching some part of 'filter'.
+   *
+   * See: https://developer.hashicorp.com/nomad/api-docs/deployments#list-deployments
+   */
+  listJobVersions: (
+    options: ListJobVersionsRequest,
+  ) => Promise<ListJobVersionsResponse>;
+};
+
+/** @public */
+export interface ListAllocationsRequest {
+  namespace: string;
+  filter: string;
+}
+
+/** @public */
+export interface ListAllocationsResponse {
+  allocations: Allocation[];
+}
+
+/** @public */
+export interface Allocation {
+  ClientStatus: string;
+  CreateTime: number;
+  DeploymentStatus: DeploymentStatus;
+  ID: string;
+  JobID: string;
+  JobVersion: string;
+  ModifyTime: number;
+  Name: string;
+  Namespace: string;
+  NodeID: string;
+  TaskGroup: string;
+}
+
+/** @public */
+export interface DeploymentStatus {
+  Healthy: boolean;
+  Timestamp: string;
+}
+
+/** @public */
+export interface ListJobVersionsRequest {
+  namespace: string;
+  jobID: string;
+}
+
+/** @public */
+export interface ListJobVersionsResponse {
+  versions: Version[];
+}
+
+/** @public */
+export interface Version {
+  ID: string;
+  SubmitTime: number;
+  Stable: boolean;
+  Version: number;
+}
+
+/** @public */
+export class FetchError extends Error {
+  get name(): string {
+    return this.constructor.name;
+  }
+
+  static async forResponse(resp: Response): Promise<FetchError> {
+    return new FetchError(
+      `Request failed with status code ${
+        resp.status
+      }.\nReason: ${await resp.text()}`,
+    );
+  }
+}
+
+/** @public */
+export class NomadHttpApi implements NomadApi {
+  static new(discoveryApi: DiscoveryApi, fetchApi: FetchApi) {
+    return new NomadHttpApi(discoveryApi, fetchApi);
+  }
+
+  constructor(private discoveryApi: DiscoveryApi, private fetchApi: FetchApi) {}
+
+  // TODO: pagination
+  async listAllocations(
+    options: ListAllocationsRequest,
+  ): Promise<ListAllocationsResponse> {
+    const apiUrl = await this.discoveryApi.getBaseUrl('nomad');
+
+    const resp = await this.fetchApi.fetch(
+      `${apiUrl}/v1/allocations?namespace=${encodeURIComponent(
+        options.namespace,
+      )}&filter=${encodeURIComponent(options.filter)}`,
+    );
+    if (!resp.ok) throw await FetchError.forResponse(resp);
+
+    return Promise.resolve({
+      allocations: await resp.json(),
+    });
+  }
+
+  // TODO: pagination
+  async listJobVersions(
+    options: ListJobVersionsRequest,
+  ): Promise<ListJobVersionsResponse> {
+    const apiUrl = await this.discoveryApi.getBaseUrl('nomad');
+
+    const resp = await this.fetchApi.fetch(
+      `${apiUrl}/v1/job/${
+        options.jobID
+      }/versions?namespace=${encodeURIComponent(options.namespace)}`,
+    );
+    if (!resp.ok) throw await FetchError.forResponse(resp);
+
+    const respJson = await resp.json();
+
+    return Promise.resolve({
+      versions: respJson.Versions,
+    });
```

Comment:
There shouldn't be a need to wrap these up in Promise.resolve as you already await things inside this function.

```suggestion
    return {
      versions: respJson.Versions,
    };
```